prov/gni/src/gnix_cq.c

```diff
@@ -495,7 +492,7 @@ static ssize_t gnix_cq_readfrom(struct fid_cq *cq, void *buf, size_t count,
 
 	fastlock_release(&cq_priv->lock);
 
-	return read_count;
+	return read_count ?: -FI_EAGAIN;
```

Comment:
just curious is the triad def such that if there is nothing before the : will it return the
read_count in this case?  I'd prefer to have it more explicit and have
```return read_count ? read_count : -FI_EAGAIN:```